velox/expression/CastExpr-inl.h

```diff
@@ -48,6 +51,61 @@ inline std::exception_ptr makeBadCastException(
       false));
 };
 
+/// @brief Convert the unscaled value of a decimal to varchar and write to raw
+/// string buffer from start position.
+/// @tparam T The type of input value.
+/// @param unscaledValue The input unscaled value.
+/// @param scale The scale of decimal.
+/// @param maxVarcharSize The estimated max size of a varchar.
+/// @param startPosition The start position to write from.
+/// @return A string view.
+template <typename T>
+StringView convertToStringView(
+    T unscaledValue,
+    int32_t scale,
+    int32_t maxVarcharSize,
+    char* const startPosition) {
+  char* writePosition = startPosition;
+  if (unscaledValue == 0) {
+    *writePosition++ = '0';
+  } else {
+    if (unscaledValue < 0) {
+      *writePosition++ = '-';
+      unscaledValue = -unscaledValue;
+    }
+    auto [position, errorCode] = std::to_chars(
+        writePosition,
+        writePosition + maxVarcharSize,
+        unscaledValue / DecimalUtil::kPowersOfTen[scale]);
+    VELOX_DCHECK_EQ(
+        errorCode,
+        std::errc(),
+        "Failed to cast decimal to varchar: {}",
+        std::make_error_code(errorCode).message());
+    writePosition = position;
+
+    if (scale > 0) {
+      *writePosition++ = '.';
+      uint128_t fraction = unscaledValue % DecimalUtil::kPowersOfTen[scale];
+      // Append leading zeros.
+      int numLeadingZeros =
+          std::max(scale - fmt::detail::count_digits(fraction), 0);
```

Comment:
Got a linter error here:

```
Call to fmt::v9::detail::count_digits is logically private, please avoid use outside of its parent namespace fbcode/velox/expression/CastExpr-inl.h Line 92
```

CC: @rui-mo

Reply:
Can we copy this function to CastExpr-inl.h?

Reply:
Sure.